### ISSUE TITLE
[feature]: Add zoom in/out in hoppscotch desktop app

### ISSUE THREAD

**Issue by mehdi-alouane** [hoppscotch/hoppscotch#3545]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The current desktop app lacks a built-in zoom function.

### Why should this be worked on?

The current desktop app lacks a built-in zoom function, which I believe is an essential feature, especially for larger monitors. Without the ability to increase the font size, this becomes even more necessary.

**Comment by vekien** [comment]
+1 for this, desperately, it's extremely hard to read on my Macbook with 2304x1440 res, it's like everything is at 8px font size. Very very difficult. Even a simple font-size option would be huge.

**Comment by liyasthomas** [comment]
Thanks for testing out our alpha release candidates of the Hoppscotch desktop app.

This is a known issue which we're currently working on. Expect this to be solved in the stable release scheduled for December.

**Comment by vieiraes** [comment]
This functionality is truly necessary for those who need to use larger fonts on the screen due to vision difficulties.

**Comment by yoratoni** [comment]
Yeah, I'm using a widescreen (21:9) and it makes Hoppscotch desktop almost unusable..

**Comment by tdharris** [comment]
I'm using a 27" 2k resolution monitor and everything is just barely too small.. can't use it. 😞 What's the status on this one?

**Comment by Gandalf-pro** [comment]
For the time being i just increased the scale factor to 2 on the .desktop entry with the env variable GDK_SCALE=2.
Its a bit bigger then i would like but fractional scaling doesn't work.

Disclaimer i am on linux and using AppImage

**Comment by szv** [comment]
Since Hoppscotch uses the relative unit `rem` in the styles, an approach could be to set the `font-size` in `:root` of the stylesheet.
Setting the font-size there to e.g. 20pt leeds to a "higher zoom/zoomed in", whereas a font-size of 10pt would be a "smaller zoom/zoomed out".

I tested it in the developer tools (Ctrl+Shift+I) in the Hoppscotch Desktop application v.2024.6.0:

## 20pt
```CSS
:root {
  font-size: 20pt;
}
```

![image](https://github.com/user-attachments/assets/4f0e16c9-afb1-4d80-a86a-29c4cad0c1f1)


## 10pt
```CSS
:root {
  font-size: 10pt;
}
```
![image](https://github.com/user-attachments/assets/840ee538-6033-404b-b74b-69607cbc5994)

**Comment by ChaitanyaPTank** [comment]
Anyupdate on this ? Where can I check progress or roadmap ?

**Comment by shellking4** [comment]
Just gave up on this and self-hosted it instead. A better option.

I've setup this docker compose inspired by what is the hoppscotch repo 
https://github.com/shellking4/hoppscotch-self-host.

When the services are up and running you can access the app on localhost:3000. Then you can install it as pwa and zoom it as you want

**Comment by ttv20** [comment]
a workaround
In linux (and mac?) you can add env to the app `.desktop` file (ussally at `~/.local/share/applications/hoppscotch.desktop`):
`Exec=env GDK_DPI_SCALE=1.2 hoppscotch`

It should work in windows too, I didn't tested it
create a shortcut to:
`cmd /c "set MY_VAR=123 && start "" the_app_path"`

**Comment by driskell** [comment]
Having this problem where the UI just gets squashed on small size screens. I'm usually able to zoom such apps so would be great to restore this. It's not just a good practice due to differing size screens but also good for accessibility.

**Comment by tdharris** [comment]
Here are alternatives for anyone else who also considers this a deal breaker:
https://openalternative.co/alternatives/postman

**Comment by kytosai** [comment]
We really need the zoom in/out feature of the application; the current UI wastes too much display area, everything is too large and takes up space unnecessarily.

**Comment by liyasthomas** [comment]
We are planning to add the zoom feature to our desktop applications soon.
Thanks for your support and patience.

**Comment by prantlf** [comment]
I use a scaled resolution. Other applications like Mail or Firefox work well. But it's next to impossible to use Hoppscotch on my Mac. It's too much eye-strain. And it's more pronounced, because Hoppscotch's shows some texts grey on white, which makes them even less readable.

* How about restoring the font size configuration, until you add the zoom feature? It should be simple, you had the feature.
* Or how about allowing a custom stylesheet to tweak the application styling? It should be a little work, no need for fancy styling/theming UI forms.

For the time being, I use Bruno instead of Hoppscotch. Black texts on white, great readability. Ctrl/Cmd+Plus works there.

**Comment by frogfreg** [comment]
developer tools (Ctrl+Shift+I) -> elements tab

``` 
body{
  zoom: 150%;
}
```

This has worked for me on the meantime

**Comment by ran222** [comment]
in the developer tools (Ctrl+Shift+I)  

var style = document.createElement('style');
style.innerHTML = ":root { font-size: 20pt; }";
document.head.appendChild(style);

**Comment by KhoaLee** [comment]
Hi, do we have any update on this issue for the zoom feature?

**Comment by Farewe1ll** [comment]
Still, we need a solution to this problem as soon as possible. While searching the web for an alternative to Postman, I didn’t expect to find an app without font size settings. I found that really frustrating, and I’m hoping for a solution.

**Comment by nicoladelbianco** [comment]
This feature would be really helpful!

**Comment by jamesgeorge007** [comment]
Apologies for the long wait on this one. PR #6358, which adds desktop zoom control in Settings, has been merged and is currently scheduled for the upcoming release.

**Comment by jamesgeorge007** [comment]
Closing since this is now addressed in the [v2026.5.0](https://github.com/hoppscotch/hoppscotch/releases/tag/2026.5.0) release.

**Comment by dungmv** [comment]
The UI size issue goes deeper than just the global zoom level. The root cause is the design system's layout density, specifically the excessive padding and massive margins on almost every element.

Even at 100%, Hoppscotch feels heavily inflated for a developer tool. Buttons, input fields, and sidebars take up way too much screen real estate, causing an unnecessary waste of space and forcing constant scrolling when working with complex APIs.

Simply scaling the webview isn't the right fix. The development team should really consider a UI/UX refactor or rewrite to tighten the layout density.

As a tool built for developers, it needs to prioritize information density over excessive whitespace. Introducing a proper "Compact Mode" (by refactoring the base Tailwind/CSS padding and spacing variables) would be a much better architectural solution than just relying on zoom configurations.

**Comment by driskell** [comment]
> As a tool built for developers, it needs to prioritize information density over excessive whitespace.

As a developer I would like to just add in that I don’t personally agree with high density layouts and tend to find them somewhat overwhelming at times. We need to remember we are all human and have differences and there are many backgrounds to cater for so no one solution fits all.

Zoom is a definite separate tool to this new discussion though and important in areas less around density and more around readability which it has solved and benefited. An additional way to modify the density would definitively be useful and lots of popular applications allow this, but that’s perhaps for a separate issue as this one has been closed.
